### PR TITLE
Add consistently check at cluster-sync

### PR DESCRIPTION
**Pull request by qinqon**
Signed-off-by: Quique Llorente <ellorent@redhat.com>

**What this PR does / why we need it**:
To check state switching after deploy, this checks that everything is good
for some time, after deploy it.

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: cluster/sync.sh [hunk 1]
```diff
@@ -77,5 +101,34 @@ for i in {300..0}; do
     exit 1
     fi
 
-    sleep 1
-done
+
+    # Deploy all needed manifests
```
Comment (phoracek) [inline cluster/sync.sh]: This is now duplicated on line 76

Comment (qinqon) [reply cluster/sync.sh]: I am blind, where it is duplicated ?

Comment (phoracek) [reply cluster/sync.sh]: https://github.com/nmstate/kubernetes-nmstate/pull/448/files/75639d9aa18d845b776fa6416c5908ae05e81280#diff-9a65a9e4d7562183f4e925310683fa9fR76

File: cluster/sync.sh [hunk 2]
```diff
@@ -2,80 +2,101 @@
...
 function isOk {
     desiredNumberScheduled=$(getDesiredNumberScheduled $1)
     numberAvailable=$(getNumberAvailable $1)
+    [ "$desiredNumberScheduled" == "$numberAvailable" ]
+}
 
-    if [ "$desiredNumberScheduled" == "$numberAvailable" ]; then
-        echo "$1 DS is ready"
-        return 0
+deploy() {
```
Comment (phoracek) [inline cluster/sync.sh]: Please follow the style of function definitions we use in the rest of the script

File: cluster/sync.sh [hunk 3]
```diff
@@ -2,80 +2,101 @@
...
-./cluster/kubectl.sh apply -f deploy/crds/nmstate.io_nodenetworkconfigurationpolicies_crd.yaml
-./cluster/kubectl.sh apply -f deploy/crds/nmstate.io_nodenetworkconfigurationenactments_crd.yaml
-if [[ "$KUBEVIRT_PROVIDER" =~ ^(okd|ocp)-.*$ ]]; then
-		./cluster/kubectl.sh apply -f deploy/openshift/
-fi
-sed \
-    -e "s#--v=production#--v=debug#" \
-    -e "s#REPLACE_IMAGE#registry:5000/nmstate/kubernetes-nmstate-handler#" \
-    deploy/operator.yaml | ./cluster/kubectl.sh create -f -
+wait_ready() {
```
Comment (phoracek) [inline cluster/sync.sh]: ditto

**Comment by qinqon** [comment]
/retest

Maybe the hostname check has to be done before running docker exec (meanikng at gocli)

**Pull request merged**